components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java

```diff
@@ -193,7 +191,9 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         }
 
         if (StringUtils.isNotEmpty(cacheEntry.getEssentialClaims())) {
-            lstEssential = getEssentialClaims(cacheEntry.getEssentialClaims());
+            essentialClaims = getEssentialClaims(cacheEntry.getEssentialClaims());
+        } else {
+            essentialClaims = new ArrayList<>();
```

Comment:
essentialClaims is already initialized. Any specific reason to re-initialize here.
I think we can skip the class level initialization of essentialClaims and use this logic

Reply:
Refactored in master.